packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts

```diff
@@ -278,6 +291,16 @@ export type ConnectButton_detailsModalOptions = {
    * thirdweb Pay allows users to buy tokens using crypto or fiat currency.
    */
   payOptions?: Extract<PayUIOptions, { mode?: "fund_wallet" }>;
+
+  /**
+   * Render custom UI for the connected wallet name in the `ConnectButton` Details Modal, overriding ENS name or wallet address.
+   */
+  connectedWalletName?: React.ReactNode;
```

Comment:
wondering if it would make more sene to let you pass a component for the entire row, so you can do your own avatar + name + balance style. would just be a single prop and gives you max flexibility. wdyt?

Reply:
@joaquim-verges that sounds great as well!

Reply:
I'd just want to be able to preserve the "switch wallet" on-click functionality at times

Reply:
hmmm in that case maybe we just expose both name and avatar as you have it, but allow for `string | React.ReactNode` for both. Let us discuss it tomorrow and we'll try to get this in asap

Reply:
Sounds good! Also [`React.ReactNode` covers `string`](https://reactnative.dev/docs/react-node), so that's why I haven't included it explicitly here.

Reply:
i think we'll go with this, but might rename to `connectedAccountName` and `connectedAccountAvatarUrl`

Reply:
Sure thing, the param naming is not important to me 🙂